### ISSUE TITLE
[Feature]: Filtering of transactions

### ISSUE THREAD

**Issue by matevz** [oasisprotocol/explorer#1636]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### SUMMARY

Add filtering capability in the list of txes

### DESCRIPTION

Suppose you have the account for which you know it deployed a contract and you want to look for its address. You open the account in the explorer, e.g. https://explorer.oasis.io/testnet/sapphire/address/0x90adE3B7065fa715c7a150313877dF1d33e777D5 and you see a bunch of transactions, most of them are contract calls.

How do I find the (latest) contract deployment transaction out of those? Ideally, the list headers should give you a drop down of possible values and you pick the one (e.g. Contract create) to filter out the results. Sorting by each column would also be nice.

**Comment by ptrus** [comment]
Yes it would make sense adding support for filtering on transaction type. Also needs to be added on nexus.

**Comment by csillag** [comment]
Designs by @donouwens:

> Desktop design: https://www.figma.com/design/8dwSyYOOm3vlgbvxPGT29p/Block-Explorer-(Post-Launch)?node-id=11727-75444&t=ZmBNUqlyYnN9EbUG-1

>Mobile design: https://www.figma.com/design/8dwSyYOOm3vlgbvxPGT29p/Block-Explorer-(Post-Launch)?node-id=11727-76179&t=ZmBNUqlyYnN9EbUG-1

**Comment by csillag** [comment]
Frontend PR is https://github.com/oasisprotocol/explorer/pull/1679.

It can be tested at https://pr-1679.oasis-explorer.pages.dev/.

**Comment by matevz** [comment]
Nice, I can confirm this works as expected!

**Comment by csillag** [comment]
> Can we also do the same for consensus? (API should already support it I believe).

Added!

**Comment by csillag** [comment]
Closed by https://github.com/oasisprotocol/explorer/pull/1679.
